datadog/data_source_datadog_security_monitoring_rules.go

```diff
@@ -0,0 +1,246 @@
+package datadog
+
+import (
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"sort"
+	"strings"
+
+	datadogV2 "github.com/DataDog/datadog-api-client-go/api/v2/datadog"
+)
+
+func dataSourceDatadogSecurityMonitoringRules() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceDatadogSecurityMonitoringRulesRead,
+
+		Schema: map[string]*schema.Schema{
+			// Filters
+			"name_filter": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "A rule name to limit the search",
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"tags_filter": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "A list of tags to limit the search",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"default_only_filter": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Limit the search to default rules",
+			},
+			"user_only_filter": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Limit the search to user rules",
+			},
```

Comment:
Two different attributes are used for boolean values to work around `GetOk` only checking if there was truthy value.

Reply:
Updated with a default value

Reply:
You're right, because you use `nil` as a possible value, so there are effectively 3 states, not just 2.